### ISSUE TITLE
Sorting in object value by customized key

### ISSUE THREAD

**Issue by congweibai** [benwinding/ngx-auto-table#29]
Hi Ben,

I wonder if tableConfig passing data with object value, for example,

data$: {
  status: [{id:'001', name: 'hello'}, {id:'002', name:'world'},{id:'003', name: 'table-package'}]
}

Is possible to set sort_by value so I can define sort_by 'id' or 'name' when I click column 'status' header?

Thanks
Wyatt

**Comment by benwinding** [comment]
Hi Wyatt 👋 
Not sure exactly what you are asking here?
The `data$` field must be an `Observable` right? So that code isn't valid. 
You can use filtering on the table already, by using the following configuration: 
``` typescript
config = {
  data$: people$,
  // Initially sort by the "name" column
  initialSort: 'name';
  // Initially sort in Descending order
  initialSortDir: "desc";
};
```
(updated docs to also show this) Is that what you were after? 🙏 
Cheers,
Ben

**Comment by congweibai** [comment]
Thanks for your reply.
I understand data$ field must be an Observable. I just try to put code simply but it makes you confused.

Similar example:

```
const people$ = of([
    {
        name: 'Wyatt',
        tag: {
          id: '001',
          position: 'developer'
      }
    },
    {
        name: 'Ben',
        tag: {
            id: '002',
            position: 'Senior developer'
        }
    },
])
config = {
  data$: people$,
  // Initially sort by the "name" column
  initialSort: 'tag'; // can we sort by 'tag.id' or 'tag.position'?
  // Initially sort in Descending order
  initialSortDir: "desc";
};
```

I am looking for if is possible to sort items by object key like 'tag.id' numerically or '' tag.postion' alphabetically in the example?

Thanks
Wyatt

**Comment by benwinding** [comment]
All good, yes I was a little confused 😅 

> I am looking for if is possible to sort items by object key like 'tag.id' numerically or '' tag.postion' alphabetically in the example?

I see now, you're right that _wasn't possible_ but I've just released a small update which allows that! 🎉 

You use it just like you said:

``` html
    <ngx-auto-table
      [config]="config"
      [columnDefinitions]="{
        name: {},
        'tag.id': {},
         ^ add sub field path here
      }"
    ></ngx-auto-table>
```

``` js
config = {
  data$: people$,
  initialSort: 'tag.id';
    ^ add matching path here too
};
```

- Available in `ngx-auto-table@10.0.27`
- Demo is here: https://benwinding.github.io/ngx-auto-table/order-demo
- Code change is here: https://github.com/benwinding/ngx-auto-table/commit/49b1d6dcc1ac1908d65545607c42ab5c2b8e8346

Let me know if there's any issues! 👌

**Comment by congweibai** [comment]
Impressive @benwinding! I like the changes and it works well.

Thanks a lot.
